src/Controllers/ApiController.php

```diff
@@ -27,4 +39,114 @@ public function index()
             ->withHeader('content-type', 'application/json')
             ->withContent(json_encode(['error' => 'Not implemented']));
     }
+
+
+    /**
+     * @return User
+     *
+     * Warning: I am no php coder. This code isnt intended for eternity but making our lives @36c3 a bit easier.
+     * Code is mostly harvested from /includes/controller/shifts_controller.php/shifts_json_export_controller()
+     */
+    protected function doApiAuth($key)
```

Comment:
Using the `$permissions` attribute would be the better option (and make the `ErrorHandler` Middleware api aware as in format the error page as JSON instead of HTML)

Reply:
$permissions gets validated before any authentication (with an api key) can happen.

Reply:
Works, somehow.
* can't get rid of doApiAuth() for getMy[Shifts|AngelTypes] because $this->user is NULL if no api_key provided (and i honestly do not understand what makes the difference here)
* the resulting 403  page will be rendered as html (with proper content_type) and I've no idea where that happens.

Reply:
That gets added by the `src/Middleware/ErrorHandler.php` (as stated above) that needs to be changed to accept json responses